pkg/cli/admin/release/extract_tools.go

```diff
@@ -632,6 +644,51 @@ const (
 	releaseVersionMarker = "!_RELEASE_VERSION_LOCATION_\x00XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\x00"
 )
 
+func getValidTarget(spec, dir, releaseName string, target extractTarget, willArchive bool) (extractTarget, bool, error) {
+	ref, err := imagereference.Parse(spec)
+	if err != nil {
+		return target, willArchive, err
+	}
+	target.Mapping.Image = spec
+	target.Mapping.ImageRef = imagesource.TypedImageReference{Ref: ref, Type: imagesource.DestinationRegistry}
+	if target.AsArchive {
+		willArchive = true
+		target.Mapping.Name = fmt.Sprintf(target.ArchiveFormat, releaseName)
+		target.Mapping.To = filepath.Join(dir, target.Mapping.Name)
+	} else {
+		target.Mapping.To = filepath.Join(dir, target.Command)
+		target.Mapping.Name = fmt.Sprintf("%s-%s", target.OS, target.Command)
+	}
+	return target, willArchive, nil
+}
+
+func verifyImageExists(opts *extract.Options, target extractTarget) error {
+	from := target.Mapping.ImageRef
+	ctx := context.Background()
+	fromContext, err := opts.SecurityOptions.Context()
+	if err != nil {
+		return err
+	}
+	fromOptions := &imagesource.Options{
+		FileDir:         opts.FileDir,
+		Insecure:        opts.SecurityOptions.Insecure,
+		RegistryContext: fromContext,
+	}
+
+	repo, err := fromOptions.Repository(ctx, from)
+	if err != nil {
+		return fmt.Errorf("unable to connect to image repository %s: %v", from.String(), err)
+	}
+	_, _, err = imagemanifest.FirstManifest(ctx, from.Ref, repo, opts.FilterOptions.Include)
+	if err != nil {
+		if imagemanifest.IsImageNotFound(err) {
+			return imagemanifest.NewImageNotFound("image does not exist", err)
+		}
+		return fmt.Errorf("unable to read image %s: %v", from, err)
+	}
```

Comment:
This error checking seems weird, given that you're not doing anything with that error I don't think that's needed. Although you're mentioning a few lines above you want to ignore not found and fallback to release image, but I'm not seeing this here.

Reply:
this is a first pass - we're checking here that the user-given reg/repo/name@sha-from-ref-image exists, ( the splice of the user-given registry/repository/name and the @sha... of the referenced release), if it doesn't exist or there's an error returning the manifest,   then default to using the referenced image - in this case would be quay.io/openshift-release-dev.. and return to the regularly scheduled program of error-checking -  (are you authorized to access the image? does the image contain the expected binary? etc) If it _does_ exist, then return to the regularly scheduled program of error-checking, also, but with the user-given-registry/repo/name@sha rather than the reference one (that won't be accessible in a disconnected env)

Reply:
So this is still WIP?

Reply:
first pass, meaning, first check the user-given registry/repo/name@sha-for-tool-image-from-ref-image and if that image exists, carry on with extracting the binary from it, but if it does not exist, then carry on with the referenced image (that may or may not be accessible in disconnected-but we're not checking that here)

Reply:
resolving this, i've pushed a change to make this cleaner